src/types/decimal/coefficient.rs

```diff
@@ -116,26 +60,26 @@ impl Coefficient {
         }
     }
 
-    /// Returns true if the Coefficient represents positive zero.
-    pub(crate) fn is_negative_zero(&self) -> bool {
-        match (self.sign, &self.magnitude.data) {
-            (Sign::Negative, UIntData::U64(0)) => true,
-            (Sign::Negative, UIntData::BigUInt(b)) if b.is_zero() => true,
-            _ => false,
-        }
+    /// Returns true if the Coefficient represents negative zero.
+    pub fn is_negative_zero(&self) -> bool {
+        self.is_zero_with_sign(Sign::Negative)
     }
 
     /// Returns true if the Coefficient represents positive zero.
-    pub(crate) fn is_positive_zero(&self) -> bool {
+    pub fn is_positive_zero(&self) -> bool {
+        self.is_zero_with_sign(Sign::Positive)
+    }
+
+    pub(crate) fn is_zero_with_sign(&self, test_sign: Sign) -> bool {
```

Comment:
🗺️ This helper method is used in the implementation of both `is_negative_zero()` and `is_positive_zero()`.

Users may wish to use `is_negative_zero()` before converting the `Coefficient` to an `Int`, as `Int` cannot represent negative zero natively. `is_positive_zero()` is provided mostly for symmetry.